git-link.el

```diff
@@ -142,10 +142,22 @@
   (save-restriction
     (widen)
     (save-excursion
-      (let ((use-region (use-region-p)))
-        (list
-         (line-number-at-pos (when use-region (region-beginning)))
-         (when use-region (line-number-at-pos (region-end))))))))
+      (let* ((use-region (use-region-p))
+             (start (when use-region (region-beginning)))
+             (end   (when use-region (region-end)))
+             (line-start (line-number-at-pos start))
+             line-end)
+        (when use-region
+          ;; If the `end' position is at the beginning of a line
+          ;; decrement the position by 1, so that the resultant
+          ;; position is on the previous line
+          (when (= end (line-beginning-position))
```

Comment:
The key is that `line-beginning-position` is compared to the value given by `region-end` (`end`).
Since `line-beginning-position` will use `point`, if `point` is above mark it's not at the end of the region and the comparison returns `nil`.